### PR TITLE
Fixes #6711: Makes kudzu check for DNR 

### LINKED ISSUE #6711: Kudzu does not respect DNR
### Labels

[BUG]

### Description

Kudzu does not respect DNR.

### Steps to reproduce

1. Die.
2. Get dragged to kudzu.
3. Set DNR because sod that.
4. Notice your body is standing up.
5. Reenter corpse.
6. Talk and move and do all sorts of things living people do.

### Expected Behavior

DNR prevents all forms of revival.

### Additional Information & Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Yellow-Mushroom**
## About the PR 

This PR makes it that, before someone becomes a kudzuperson, they have their DNR status checked. 

If they have set DNR, they will be gibbed right before the bulb is opened, having the bulb open to a pretty grotesque scene with some flavor text. 

(I can add a video if requested)

The player has the entire time they are within a cocoon, to the last moments of being in the bulb to set DNR

## Why's this needed? 

DNR should be respected

fixes #6711

### CODE REVIEW

File: code/modules/events/gimmick/kudzu.dm [hunk 1]
```diff
@@ -403,6 +403,13 @@
 
 				sleep(bulb_complete)
 
+				if(!isalive(M) && M.ghost && M.ghost.mind && M.ghost.mind.dnr)
+					src.visible_message("<span class='alert'>The bulb opens, having drained all the nutrients from [M.name]!</span>")
+					M.gib()
+					flick("bulb-open-animation", src)
+					new/obj/decal/opened_kudzu_bulb(get_turf(src.loc))
+					sleep(10)
```
Comment (Zonespace27) [inline code/modules/events/gimmick/kudzu.dm:411]:
```suggestion
					sleep(1 SECOND)
```

**Comment by Zonespace27** [comment]
Put "Fixes #6711" in PR body for autoclose of the issue

File: code/modules/events/gimmick/kudzu.dm [hunk 2]
```diff
@@ -403,6 +403,13 @@
 
 				sleep(bulb_complete)
 
+				if(!isalive(M) && M.ghost && M.ghost.mind && M.ghost.mind.dnr)
```
Comment (ZeWaka) [inline code/modules/events/gimmick/kudzu.dm:406]: try using `?.` syntax here :)

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/modules/events/gimmick/kudzu.dm (hunk 2 above)
Comment (Yellow-Mushroom) [reply code/modules/events/gimmick/kudzu.dm]: should be fixed by 7e3c12c

File: code/modules/events/gimmick/kudzu.dm [hunk 3]
```diff
@@ -403,7 +403,15 @@
 
 				sleep(bulb_complete)
 
-				if (!destroyed && ishuman(M))
+				if(!isalive(M) && M.ghost?.mind?.dnr)
+					src.visible_message("<span class='alert'>The bulb opens, having drained all the nutrients from [M.name]!</span>")
```
Comment (ZeWaka) [inline code/modules/events/gimmick/kudzu.dm:407]:
```suggestion
					src.visible_message("<span class='alert'>The [src] opens, having drained all the nutrients from [M]!</span>")
```

File: code/modules/events/gimmick/kudzu.dm [hunk 4]
```diff
@@ -403,7 +403,15 @@
...
+					M.gib()
+					flick("bulb-open-animation", src)
+					new/obj/decal/opened_kudzu_bulb(get_turf(src.loc))
```
Comment (ZeWaka) [inline code/modules/events/gimmick/kudzu.dm:410]: unnecessary
```suggestion
					new/obj/decal/opened_kudzu_bulb(get_turf(src))
```

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**